pkg/agent/apiserver/handlers/networkpolicy/handler.go

```diff
@@ -17,51 +17,64 @@ package networkpolicy
 import (
 	"encoding/json"
 	"net/http"
+	"net/url"
 
-	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
+	agentquerier "github.com/vmware-tanzu/antrea/pkg/agent/querier"
 	cpv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/querier"
 )
 
 // HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
 // to query network policy rules in current agent.
-func HandleFunc(aq querier.AgentQuerier) http.HandlerFunc {
+func HandleFunc(aq agentquerier.AgentQuerier) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		name := r.URL.Query().Get("name")
-		ns := r.URL.Query().Get("namespace")
-		pod := r.URL.Query().Get("pod")
-
-		if (name != "" || pod != "") && ns == "" {
-			http.Error(w, "namespace must be provided", http.StatusBadRequest)
-			return
-		}
+		npFilter := NewFilterFromURLQuery(r.URL.Query())
 
 		var obj interface{}
 		npq := aq.GetNetworkPolicyInfoQuerier()
+		var nps []cpv1beta1.NetworkPolicy
 
-		if name != "" {
-			// Query the specified NetworkPolicy.
-			np := npq.GetNetworkPolicy(name, ns)
-			if np != nil {
-				obj = *np
-			}
-		} else if pod != "" {
-			// Query NetworkPolicies applied to the Pod
-			interfaces := aq.GetInterfaceStore().GetContainerInterfacesByPod(pod, ns)
-			if len(interfaces) > 0 {
-				nps := npq.GetAppliedNetworkPolicies(pod, ns)
-				obj = cpv1beta1.NetworkPolicyList{Items: nps}
+		if npFilter.Pod != "" {
+			if npFilter.Namespace == "" {
+				http.Error(w, "With a pod name, namespace must be provided", http.StatusBadRequest)
+				return
+			} else {
+				interfaces := aq.GetInterfaceStore().GetContainerInterfacesByPod(npFilter.Pod, npFilter.Namespace)
+				if len(interfaces) > 0 {
+					nps = npq.GetAppliedNetworkPolicies(npFilter.Pod, npFilter.Namespace, *npFilter)
+				}
 			}
 		} else {
-			nps := npq.GetNetworkPolicies(ns)
-			obj = cpv1beta1.NetworkPolicyList{Items: nps}
+			nps = npq.GetNetworkPolicies(*npFilter)
 		}
 
-		if obj == nil {
-			w.WriteHeader(http.StatusNotFound)
-			return
-		}
+		obj = cpv1beta1.NetworkPolicyList{Items: nps}
+
 		if err := json.NewEncoder(w).Encode(obj); err != nil {
 			http.Error(w, "Failed to encode response: "+err.Error(), http.StatusInternalServerError)
 		}
 	}
 }
+
+// From user shorthand input to cpv1beta1.NetworkPolicyType
+var mapToNetworkPolicyType = map[string]cpv1beta1.NetworkPolicyType{
+	"np":    cpv1beta1.K8sNetworkPolicy,
+	"NP":    cpv1beta1.K8sNetworkPolicy,
+	"k8snp": cpv1beta1.K8sNetworkPolicy,
+	"K8snp": cpv1beta1.K8sNetworkPolicy,
+	"K8sNP": cpv1beta1.K8sNetworkPolicy,
+	"cnp":   cpv1beta1.AntreaClusterNetworkPolicy,
```

Comment:
Let's update these to ACNP now

Reply:
Also for this map we could use ToUpper(userInput) for instance to get rid of similar entries?

Reply:
> Also for this map we could use ToUpper(userInput) for instance to get rid of similar entries?

Good point.